services/device-registry-file/src/main/java/org/eclipse/hono/deviceregistry/file/AbstractFileBasedRegistryConfigProperties.java

```diff
@@ -164,4 +168,34 @@ public final boolean isStartEmpty() {
     public final void setStartEmpty(final boolean flag) {
         this.startEmpty = flag;
     }
+
+    /**
+     * Get the list of authorised hashing algorithms for already hashed passwords.
+     * The device registry will not accept credentials using a hashing
+     * algorithm that is not present in that list.
+     * If the list is empty, the device registry will accept any hashing algorithm.
+     * <p>
+     * Default value is an empty HashSet.
+     *
+     * @return The list of authorized algorithms.
+     */
+    public Set<String> getHashAlgorithmsWhitelist() {
+        return hashAlgorithmsWhitelist;
+    }
+
+    /**
+     * Set the list of authorised hashing algorithms for already hashed passwords.
+     * The device registry will not accept credentials using a hashing
+     * algorithm that is not present in that list.
+     * <p>
+     * The default value is empty.
+     *
+     * @param hashAlgorithmsWhitelist The list of authorized algorithms.
+     */
+    public void setHashAlgorithmsWhitelist(final String[] hashAlgorithmsWhitelist) {
```

Comment:
We should probably use set instead of array here as well?

Reply:
springboot create an array from a YAML list, or comma separated values in an environment variable :)